model/span.go

```diff
@@ -28,6 +28,25 @@ var (
 	ErrValidDurationRequired = errors.New("valid duration required")
 )
 
+// Baggage holds the interface for interacting with baggage context propagation
+// implementations.
+type Baggage interface {
+	// Init returns a fresh Baggage primed for usage in a request lifecycle.
+	Init() Baggage
+	// AddHeader adds the provided values to a header designated by key. If key
+	// is not in the whitelist it will be a noop and return false.
+	AddHeader(key string, value ...string) bool
+	// DeleteHeader removes the header data designated by key. If key is not
+	// in the whitelist it will be a noop and return false.
+	DeleteHeader(key string) bool
+	// IterateHeaders will iterate over the whitelist headers and for each found
+	// header with data it will trigger the callback function.
+	IterateHeaders(f func(key string, values []string))
+	// IterateWhiteList iterates over the whitelist headers and trigger the
+	// callback function providing it with each header key.
+	IterateWhiteList(f func(key string))
```

Comment:
I have two comments here:

1. I really dislike the `WhiteList` name, I know this is the usual practice but harmless terms like `allowList` have got some traction now, enough to consider changing this.
2. I'd like to understand the use case where this function is needed. Like the use case where the user needs to iterate through this keys even when the `IterateHeaders` provide the similar output.